platform/admin/core/admin-core-impl/src/main/java/org/codice/ddf/admin/core/config/mapping/ProfileConfigMappingProvider.java

```diff
@@ -0,0 +1,184 @@
+    try (InputStream inputStream = new FileInputStream(profilesFilePath)) {
+      String json = IOUtils.toString(inputStream, StandardCharsets.UTF_8);
+      LOGGER.debug("profiles.json: {}", json);
+      this.profiles = (Map<String, Object>) Boon.fromJson(json);
+      LOGGER.debug("profiles: {}", this.profiles);
+      processProfileConfigSections();
+      LOGGER.debug("configs: {}", configs);
+      LOGGER.debug("pids: {}", pids);
+    } catch (IOException e) {
+      LOGGER.debug(
+          "Could not find {} during installation. Using default profile.", profilesFilePath, e);
+      this.profiles = Collections.EMPTY_MAP;
+    }
+  }
+
+  @Override
+  public boolean isPartial() {
+    return true;
+  }
+
+  @Override
+  public boolean canProvideFor(ConfigMapping.Id id) {
+    LOGGER.debug(
+        "Received {}; can provide for: {}; returning: {}",
+        id.getName(),
+        pids.toString(),
+        pids.contains(id.getName()));
+    return pids.contains(id.getName());
+  }
+
+  @Override
+  public Map<String, Object> provide(ConfigMapping.Id id, ConfigService configService)
+      throws ConfigMappingException {
+    LOGGER.debug("Providing for: {}", id.getName());
+    final Optional<NetworkProfileConfig> optionalNetworkProfileConfig =
+        configService.get(NetworkProfileConfig.class);
+
+    if (!optionalNetworkProfileConfig.isPresent()) {
+      return Collections.EMPTY_MAP;
+    }
+
+    final NetworkProfileConfig networkProfileConfig = optionalNetworkProfileConfig.get();
+    final String profile = networkProfileConfig.getProfile();
+    final String version = networkProfileConfig.getVersion();
+    LOGGER.debug("profile: {}", profile);
+    LOGGER.debug("version: {}", version);
+
+    final List<ConfigEntry> profileConfigs = configs.get(profile);
+    LOGGER.debug("configs for profile [{}]: {}", profile, profileConfigs);
+    final Optional<ConfigEntry> optConfigEntry =
+        profileConfigs.stream().filter(c -> c.hasPid(id.getName())).findFirst();
+
+    if (!optConfigEntry.isPresent()) {
+      return Collections.EMPTY_MAP;
+    }
+
+    final ConfigEntry configEntry = optConfigEntry.get();
+    final Map<String, Object> properties = configEntry.getProperties();
+    LOGGER.debug("Providing mappings: {}", properties);
+    return properties;
+  }
+
+  private void processProfileConfigSections() {
+    profiles.entrySet().forEach(this::processProfileConfigs);
+  }
+
+  private void processProfileConfigs(Map.Entry<String, Object> profileEntry) {
+    final String profile = profileEntry.getKey();
+    LOGGER.debug("Process config sections for profile [{}].", profile);
+    final List<Map<String, Object>> profileConfigs = getConfigsFromProfileEntry(profileEntry);
+    LOGGER.debug("profile configs: {}", profileConfigs);
+    final List<ConfigEntry> configEntries =
+        profileConfigs.stream().map(ConfigEntry::new).collect(Collectors.toList());
+    configs.put(profile, configEntries);
+    Set<String> pidsSet =
```

Comment:
why create a temporary set just to re-add all entries to the instance attribute one? Simply collect them into a set that you can assign directly and avoid the temporary collection creation.

Reply:
fixed